torchtext/experimental/transforms.py

```diff
@@ -0,0 +1,81 @@
+import torch
+import torch.nn as nn
+from torchtext.experimental import functional as F
+from typing import List, Tuple
+
+
+__all__ = [
+    'BasicEnglishNormalize',
+    'RegexTokenizer'
+]
+
+
+class BasicEnglishNormalize(nn.Module):
+    r"""Basic normalization for a string sentence.
+
+    Normalization includes
+    - lowercasing
+    - complete some basic text normalization for English words as follows:
+        add spaces before and after '\''
+        remove '\"',
+        add spaces before and after '.'
+        replace '<br \/>'with single space
+        add spaces before and after ','
+        add spaces before and after '('
+        add spaces before and after ')'
+        add spaces before and after '!'
+        add spaces before and after '?'
+        replace ';' with single space
+        replace ':' with single space
+        replace multiple spaces with single space
+    """
+
+    def __init__(self):
+        super(BasicEnglishNormalize, self).__init__()
+        self.patterns_list = [
+            (r'\'', ' \'  '),
+            (r'\"', ''),
+            (r'\.', ' . '),
+            (r'<br \/>', ' '),
+            (r',', ' , '),
+            (r'\(', ' ( '),
+            (r'\)', ' ) '),
+            (r'\!', ' ! '),
+            (r'\?', ' ? '),
+            (r'\;', ' '),
+            (r'\:', ' '),
+            (r'\s+', ' ')]
+
+    def forward(self, line: str) -> List[str]:
+        r"""
+        Args:
+            line (str): a line of text to tokenize.
+        Returns:
+            List[str]: a list of tokens after normalizing and splitting on whitespace.
+        """
+        return F.regex_tokenizer(line, self.patterns_list)
+
+
+class RegexTokenizer(nn.Module):
+    r"""Regex tokenizer for a string sentence.
+
+    Steps include
+    - lowercasing
+    - applying all regex replacements defines in patterns_list
+
+    Args:
+        patterns_list (List[Tuple[str, str]]): a list of tuples (ordered pairs) which contain the regex pattern string as the first element and the replacement string as the second element.
```

Comment:
You can include an example to use the func/class, like this
https://github.com/pytorch/text/blob/e70955309ead681f924fecd36d759c37e3fdb1ee/torchtext/experimental/datasets/text_classification.py#L137-L143